lib/deploy.go

```diff
@@ -15,17 +16,47 @@ import (
 // and then deploys the package to salesforce
 func PushByPaths(fpaths []string, byName bool, namePaths map[string]string, opts *ForceDeployOptions) {
 	pb := NewPushBuilder()
-
 	var badPaths []string
 	for _, fpath := range fpaths {
-		// TODO: check for folder, if a folder, add all files in it
-		name, err := pb.AddFile(fpath)
+
+		fi, err := os.Stat(fpath)
 		if err != nil {
-			fmt.Println(err.Error())
+			fmt.Println(err)
 			badPaths = append(badPaths, fpath)
-		} else {
-			// Store paths by name for error messages
-			namePaths[name] = fpath
+			continue
+		}
+
+		mode := fi.Mode()
+		//If path provided is dir we are adding all containing files to deployment
+		if mode.IsDir() {
+			files, err := ioutil.ReadDir(fpath)
+			if err != nil {
+				fmt.Println(err)
+				badPaths = append(badPaths, fpath)
+			}
+
+			for _, f := range files {
+				if f.IsDir() {
+					continue
```

Comment:
I think pushing a directory should always be recursive.  For example, `force push src/` and `force push src/reports/` should work.